datafusion/functions/src/string/starts_with.rs

```diff
@@ -81,18 +106,73 @@ impl ScalarUDFImpl for StartsWithFunc {
     }
 
     fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
-        use DataType::*;
-
-        Ok(Boolean)
+        Ok(DataType::Boolean)
     }
 
     fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
         match args[0].data_type() {
             DataType::Utf8 => make_scalar_function(starts_with::<i32>, vec![])(args),
-            DataType::LargeUtf8 => {
-                return make_scalar_function(starts_with::<i64>, vec![])(args);
-            }
-            _ => internal_err!("Unsupported data type"),
+            DataType::LargeUtf8 => make_scalar_function(starts_with::<i64>, vec![])(args),
+            DataType::Utf8View => make_scalar_function(starts_with::<i32>, vec![])(args),
+            _ => internal_err!("Unsupported data types for starts_with")?,
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::utils::test::test_function;
+    use arrow::array::{Array, BooleanArray};
+    use arrow::datatypes::DataType::Boolean;
+    use datafusion_common::{Result, ScalarValue};
+    use datafusion_expr::{ColumnarValue, ScalarUDFImpl};
+
+    use super::*;
+
+    #[test]
+    fn test_functions() -> Result<()> {
+        // Generate test cases for starts_with
+        let test_cases = vec![
```

Comment:
Not sure if this is the best way to test this change. It tests the function but doesn't a view wouldn't be coerced.

Reply:
I have tests for the coercion in https://github.com/apache/datafusion/pull/11753 -- you beat me to filing the ticket  ❤️  -- I plan to make an epic / start filing items for the various other functions

Reply:
I recommend using a slt test (example of sql test above)

Reply:
Just to throw it out there, if you make the top level epic, I'd be happy to work though that PR and make individual issues for the problematic functions.

Reply:
Deal! I filed https://github.com/apache/datafusion/issues/11790

What I recommend doing is starting with a single ticket and making it a complete template / description with the idea being that that someone that is not familar DataFusion or Utf8View can work on it

Then when we file tickets for the rest of the functions we can reuse the same description over again. This
1. Reduces the barrier to contribution
2. I have found people love working on the code when there is a clear description

I took a shot at updating https://github.com/apache/datafusion/issues/11786 with background for use a template. Let me know what you think

Reply:
Sounds good to me. I'll have a go at replicating one of the other functions following #11786, and then if we're happy with it, I can grind through the others.

Reply:
Sounds good -- thank you. What I have found works well in the past is to do like 10 at a time (if you make more than that the number of parallel threads / work I found overwhelmed our ability to review them)